### PR TITLE
fix(JOHNZON-397): protection against very slow BigDecimal to BigInteger conversion at large scale

### PR DESCRIPTION
**Pull request by jeanouii**
Requires maybe an option to configure the limit. Not sure if anyone needs something bigger than 1k digits.

### CODE REVIEW

File: johnzon-core/src/main/java/org/apache/johnzon/core/JsonNumberImpl.java [hunk 1]
```diff
@@ -117,4 +119,17 @@ private void checkFractionalPart() {
             throw new ArithmeticException("Not an int/long, use other value readers");
         }
     }
+
+    private void checkBigIntegerScale() {
```
Comment (rmannibucau) [inline johnzon-core/src/main/java/org/apache/johnzon/core/JsonNumberImpl.java:123]: guess we need a toggle in the `Map` + `JsonbConfig` + `MapperBuilder` to disable it for backward compat - it stays legal?

File: johnzon-core/src/test/java/org/apache/johnzon/core/JsonNumberTest.java [hunk 2]
```diff
@@ -62,20 +64,62 @@ public void equals() {
...
+
     @Test
     public void testBigInteger() {
-       
         JsonArray array = Json.createArrayBuilder().add(100.0200).build();
         Assert.assertEquals(new BigInteger("100"), array.getJsonNumber(0).bigIntegerValue());
+    }
+
+    @Test
+    public void testSlowBigIntegerConversion() {
```
Comment (rmannibucau) [inline johnzon-core/src/test/java/org/apache/johnzon/core/JsonNumberTest.java:78]: maybe test failing cases?

**Review by rmannibucau — CHANGES_REQUESTED** [review]
Ok with a toggle + test for the error case to me

File: johnzon-core/src/test/java/org/apache/johnzon/core/JsonNumberTest.java (hunk 2 above)
Comment (jeanouii) [reply johnzon-core/src/test/java/org/apache/johnzon/core/JsonNumberTest.java:78]: It's right bellow

**Review by jgallimore — APPROVED** [review]

**Pull request merged**